Winch/Core/API/Events/Addressables/AddressablesLoadedHook.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine.ResourceManagement.AsyncOperations;
+
+namespace Winch.Core.Events.Addressables
+{
+    public class AddressablesLoadedHook<T>
+    {
+        public event AddressablesLoadedEventHandler<T> Before;
+        public event AddressablesLoadedEventHandler<T> On;
+
+        public void Trigger(object sender, AsyncOperationHandle<IList<T>> handle, bool prefix)
+        {
+            WinchCore.Log.Debug($"Triggered {typeof(T)} type event: {handle.Result.Count} elements (Prefix: {prefix})");
```

Comment:
This will look like: 
> Triggered AchievementData type event: 40 elements (Prefix: True)
which may be less desirable than something like
> Triggered AchievementDataLoaded type event: 40 elements (Prefix: True)

Reply:
The difference in logging there is negligible, and the elegance of the solution is absolutely worth it imo.